src/App.vue

```diff
@@ -15,21 +22,139 @@ export default {
 
   data() {
     return {
-      messages: []
+      intervalId: null,
+      isLoading: false,
+      messages: [],
+      refreshInterval: 30
     };
   },
 
   methods: {
+    startUpdateMessages() {
+      this.intervalId = setInterval(() => {
+        this.updateMessages();
+      }, this.refreshInterval * 1000);
+    },
     updateMessages() {
-      api.getMessages().then(messages => (this.messages = messages));
+      this.isLoading = true;
+      this.messages = [];
+      api.getMessages().then(messages => {
+        this.messages = messages;
+        this.isLoading = false;
+      });
+    },
+    updateRefreshInterval(newRefreshInterval) {
+      this.refreshInterval = newRefreshInterval;
+      clearInterval(this.intervalId);
+      if (this.refreshInterval) this.startUpdateMessages();
```

Comment:
I don't like this syntax, the curly braces are good for readability. add `'curly': 'error'` to .eslintrc.js rules. Could be nice to add a github action which check the style is good with eslint (maybe create an issue).